spec/javascripts/app/views/location_stream_spec.js

```diff
@@ -0,0 +1,39 @@
+describe("app.views.LocationMap", function() {
+  beforeEach(function(){
+    this.post = factory.post();
+    this.view = new app.views.LocationStream({model : this.post});
+  });
+
+  describe("toggleMap", function() {
+    context("with location provided", function() {
+      beforeEach(function(){
+        this.post.set({location : factory.location()}); // set location
+        gon.appConfig = { map: {enabled: true, mapbox: {enabled: true, id: "yourID", accessToken: "yourAccessToken" }} };
+        this.view.render();
+      });
+
+      it("should contain a map container", function() {
+        expect(this.view.$el[0]).toContainElement(".mapContainer");
+      });
+
+      it("should initialize map", function() {
+        expect(this.view.$el.find(".mapContainer")[0]).toHaveClass("empty");
+        this.view.toggleMap();
+        expect(this.view.$el.find(".mapContainer")[0]).not.toHaveClass("empty");
+      });
+
+      it("should change display status on every click", function() {
```

Comment:
It looks like the jQuery function `toggle` only works if the element has been added to the pages html, so I added the rendered html to `spec.content()` which is element where you want to add your html inside of a jasmine test. I also renamed the spec to `app.views.LocationStream`, since this is the name of the view you are testing, and unified the tests afterwards. Here is the diff:
```diff
diff --git a/spec/javascripts/app/views/location_stream_spec.js b/spec/javascripts/app/views/location_stream_spec.js
index 09186bc..055541c 100644
--- a/spec/javascripts/app/views/location_stream_spec.js
+++ b/spec/javascripts/app/views/location_stream_spec.js
@@ -1,38 +1,43 @@
-describe("app.views.LocationMap", function() {
+describe("app.views.LocationStream", function() {
   beforeEach(function(){
     this.post = factory.post();
     this.view = new app.views.LocationStream({model : this.post});
+    gon.appConfig = { map: {enabled: true, mapbox: {enabled: true, id: "yourID", accessToken: "yourAccessToken" }} };
   });
 
   describe("toggleMap", function() {
     context("with location provided", function() {
       beforeEach(function(){
         this.post.set({location : factory.location()}); // set location
-        gon.appConfig = { map: {enabled: true, mapbox: {enabled: true, id: "yourID", accessToken: "yourAccessToken" }} };
-        this.view.render();
+        spec.content().html(this.view.render().el);
       });
 
       it("should contain a map container", function() {
-        expect(this.view.$el[0]).toContainElement(".mapContainer");
+        expect(spec.content()).toContainElement(".mapContainer");
       });
 
       it("should initialize map", function() {
-        expect(this.view.$el.find(".mapContainer")[0]).toHaveClass("empty");
+        expect($(".mapContainer")).toHaveClass("empty");
         this.view.toggleMap();
-        expect(this.view.$el.find(".mapContainer")[0]).not.toHaveClass("empty");
+        expect($(".mapContainer")).not.toHaveClass("empty");
       });
 
       it("should change display status on every click", function() {
         this.view.toggleMap();
-        expect(this.view.$el.find(".mapContainer")[0]).toHaveCss({display: "block"});
+        expect($(".mapContainer")).toHaveCss({display: "block"});
         this.view.toggleMap();
-        expect(this.view.$el.find(".mapContainer")[0]).toHaveCss({display: "none"});
+        expect($(".mapContainer")).toHaveCss({display: "none"});
       });
 
-    }),
+    });
+
     context("without location provided", function() {
+      beforeEach(function(){
+        spec.content().html(this.view.render().el);
+      });
+
       it("should not initialize the map", function() {
-        expect(this.view.$el[0]).not.toContainElement(".mapContainer");
+        expect(spec.content()).not.toContainElement(".mapContainer");
       });
     });
   });
```